src/runtime_src/driver/xclng/xrt/util/message.cpp

```diff
@@ -144,40 +167,86 @@ syslog_dispatch::~syslog_dispatch() {
 }
 
 void
-syslog_dispatch::send(severity_level l, const char* msg) {
+syslog_dispatch::send(severity_level l, const char* tag, const char* msg) {
   syslog(severityMap[l], "%s", msg);
 }
 
 //file ops
 file_dispatch::file_dispatch(const std::string &file) {
   handle.open(file.c_str());
+  handle << "XRT build version: " << xrt_build_version << std::endl;
+  handle << "Build hash: " << xrt_build_version_hash << std::endl;
+  handle << "Build date: " << xrt_build_version_date << std::endl;
+  handle << "Git branch: " << xrt_build_version_branch<< std::endl;
+  //timestamp
+  auto time = std::chrono::system_clock::now();
```

Comment:
Lets wrap this chrono call and conversion to string in a single function.  
Copy runtime_src/xrt/util/time.{h,cpp} to xclng/xrt/util.h (change namespace to xrt_core) and add the function you need.